cocos/core/math/quat.ts

```diff
@@ -908,7 +916,7 @@ export function quat (other: Quat): Quat;
 export function quat (x?: number, y?: number, z?: number, w?: number): Quat;
 
 export function quat (x: number | Quat = 0, y = 0, z = 0, w = 1): Quat {
-    return new Quat(x as any, y, z, w);
+    return new Quat(x as number, y, z, w);
```

Comment:
Can x b a Quat instance?

Reply:
The constructor of Quat will check that, here just do a trick for ts compiler.